src/NetworkManager.cpp

```diff
@@ -144,114 +54,88 @@ void NetworkManager::disconnect()
     DBG("WiFi Disconnected.");
 }
 
-void NetworkManager::taskScanWifi(void *pvParameters)
+void NetworkManager::_handleOTA(TimerHandle_t handle)
 {
-    NetworkManager *self = (NetworkManager *)pvParameters;
+    ArduinoOTA.handle();
+}
 
-    for (;;)
-    {
-        xEventGroupWaitBits(self->_eg, EVENT_SCAN_WIFI, pdTRUE, pdTRUE, portMAX_DELAY);
+void NetworkManager::_blinkStatusLed(TimerHandle_t handle)
+{
+    IO *io = IO::getInstance();
+    io->set(OSK_STATUS_LED, !io->get(OSK_STATUS_LED));
+}
 
-        WiFi.mode(WIFI_STA);
+void NetworkManager::beginOTA(std::string hostname)
+{
+    if (hostname.length() > 0) {
+        ArduinoOTA.setHostname(hostname.c_str());
+    }
+    ArduinoOTA.begin();
 
-        DBG("Start WiFi scanning...");
-        
-        int n = WiFi.scanNetworks();
+    _tOTA = xTimerCreate("ota", pdMS_TO_TICKS(2000), pdTRUE, (void *)0, reinterpret_cast<TimerCallbackFunction_t>(_handleOTA));
+	xTimerStart(_tOTA, 0);
+}
+
+void NetworkManager::useStatusLedForWiFi()
+{
+    IO *io = IO::getInstance();
+	io->mode(OSK_STATUS_LED, OUTPUT);
+	io->set(OSK_STATUS_LED, LOW);
 
-        DBG("Found WiFis: ");
+    _tStatus = xTimerCreate("statusBlink", pdMS_TO_TICKS(STATUS_BLINK_WIFI_DISABLED), pdTRUE, (void *)0, reinterpret_cast<TimerCallbackFunction_t>(_blinkStatusLed));
+	xTimerStart(_tStatus, 0);
+}
 
-        NimBLEService *pSvc = self->bleServer->getServiceByUUID(BLE_SERVICE_MAIN_ADDR);
-        NimBLECharacteristic *pChr = pSvc->getCharacteristic(BLE_CHAR_SCAN_WIFI_ADDR);
+void NetworkManager::_WiFiStationConnected()
+{
+    if (_tStatus) {
+        xTimerChangePeriod(_tStatus, pdMS_TO_TICKS(STATUS_BLINK_WIFI_CONNECTED), portMAX_DELAY);
+    }
 
-        for (int i = 0; i < n; ++i)
-        {
-            std::string wifiName(WiFi.SSID(i).c_str());
-            DBG(wifiName.c_str());
-            pChr->setValue(wifiName);
-            pChr->notify();
-            delay(10);
-        }
+    if (_isSmartConfigStarted && WiFi.status() == WL_CONNECTED) {
+        char buffer[32];
+        
+		strlcpy(buffer, WiFi.SSID().c_str(), sizeof(buffer));
+        DBG("Save WfFi: %s", buffer);
+        _settings->setWifiName(buffer);
 
-        pChr->setValue('+');
-        pChr->notify();
+		strlcpy(buffer, WiFi.psk().c_str(), sizeof(buffer));
+        DBG("Save PWD: %s", buffer);
+        _settings->setWifiPassword(buffer);
 
-        DBG("Done WiFi scanning.");
+        _isSmartConfigStarted = false;
     }
+
+    DBG("WiFi is connected!");
+    DBG("IP: %s", WiFi.localIP().toString().c_str());
+
+    WiFi.setAutoReconnect(true);
 }
 
-void NetworkManager::WiFiEvent(WiFiEvent_t event)
+void NetworkManager::_WiFiEvents(WiFiEvent_t event)
 {

```

Comment:
It'll be good (in the future) to allow user handlers for connect/disconnect events.